pulsar-broker-common/src/main/java/org/apache/pulsar/common/configuration/VipStatus.java

```diff
@@ -47,13 +47,33 @@ public class VipStatus {
     // log a full thread dump when a deadlock is detected in status check once every 10 minutes
     // to prevent excessive logging
     private static final long LOG_THREADDUMP_INTERVAL_WHEN_DEADLOCK_DETECTED = 600000L;
-    private static volatile long threadDumpLoggedTimestamp;
+    private static volatile long lastCheckStatusTimestamp;
+
+    // Since the status endpoint doesn't have authentication, it will be necessary to have a solution to prevent
+    // introducing a new DoS vulnerability where calling the status endpoint in a tight loop could introduce
+    // significant load to the system. One way would be to check that the deadlock check is executed only
+    // when there's more than 1 seconds from the previous check.
+    // If it's less than that, the previous result of the deadlock check would be reused.
+    private static final long DEADLOCK_DETECTED_INTERVAL = 1000L;
+    private static volatile boolean brokerIsHealthy = true;
```

Comment:
I'd suggest renaming this field so that it's more specific
```suggestion
    private static volatile boolean lastCheckStatusResult;
```